src/main/java/com/coderace/service/DeliveryService.java

```diff
@@ -0,0 +1,58 @@
+package com.coderace.service;
+
+import com.coderace.delivery.Delivery;
+import com.coderace.model.enums.DeliveryType;
+import com.coderace.model.dtos.DeliveryRequestDTO;
+import com.coderace.model.dtos.DeliveryResponseDTO;
+import com.coderace.model.exceptions.BadRequestException;
+import com.coderace.repository.DeliveryRepository;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class DeliveryService {
+    private final DeliveryRepository repository;
+
+    public DeliveryService(DeliveryRepository repository) {
+        this.repository = repository;
+    }
+
+    public DeliveryResponseDTO create(DeliveryRequestDTO requestDTO) {
+        final DeliveryType type = resolveType(requestDTO.getType());
+        final String code = resolveCode(requestDTO.getCode());
+
+        final Delivery delivery = new Delivery(code, type);
+
+        repository.save(delivery);
+        return buildDeliveryResponseDTO(delivery);
+    }
```

Comment:
Cómo resolver test parte 2:
La clave del error es que en la línea 27, cuando llamás al buildDeliveryResponseDTO, no estás enviando _exactamente lo que devuelve el repository_, sino que estás usando el delivery que ya está inicializado en la línea 24 (el cual no está mockeado). Para resolverlo deberías asegurarte que en la línea 27 se use lo que devuelve el repository, es decir, tu mock:

![image](https://user-images.githubusercontent.com/94071797/145226327-44028b7e-74a7-41fe-9979-98184f3a34e3.png)

Reply:
By the way, aplicaría este mismo tratamiento en el create() de ExampleService ya que está fallando el test por lo mismo